### ISSUE TITLE
Cl Experiments On OpeNTF

### ISSUE THREAD

**Issue by rezaBarzgar** [fani-lab/OpeNTF#212]
This issue is for logging the results of the OpeNTF with 2 different loss-based curriculum learning methods (`Data Parameters`, `SuperLoss`) on 4 datasets (`IMDb`, `GitHub`, `DBLP`, `USPT`) by 2 models that have state-of-the-art results (`fnn_emb`, `bnn_emb`).

In These experiments, hyperparameters should remain the same. 
- Learning rate: `0.001`
- epoch: `10`
- batch size: `128`
- Early Stopping: `True` with patience `5`
- weight decay in Cl methods: `0.9`

**Comment by rezaBarzgar** [comment]
@hosseinfani 
Dr. Fani,

If you approve, I plan to assign this task to Marco. I will ask him to run it on Compute Canada and share the results with us. I believe it will be a good learning experience for Marco to become familiar with the process of running large-scale projects on cloud resources. I appreciate your thoughts on this.

Thank you

**Comment by hosseinfani** [comment]
@rezaBarzgar 
perfect. thank you.

**Comment by MarcoKurepa** [comment]
On it!

**Comment by MarcoKurepa** [comment]
@hosseinfani Please confirm me on Compute Canada.
![image](https://github.com/fani-lab/OpeNTF/assets/105560378/5927df6b-b71c-408b-85b9-ce77754d0516)

**Comment by hosseinfani** [comment]
@MarcoKurepa 
already done!

**Comment by MarcoKurepa** [comment]
Response from Compute Canada support is already in. 
[Gmail - Error 4 on Cluster Beluga.pdf](https://github.com/fani-lab/OpeNTF/files/12242492/Gmail.-.Error.4.on.Cluster.Beluga.pdf)

**Comment by MarcoKurepa** [comment]
https://sharcnet-ca.zoom.us/meeting/register/tZ0rduuspzktH92Cq3-9AWe-HaUBmPyCBe78

**Comment by MarcoKurepa** [comment]
I changed ```computecanada.sh``` by removing ```#SBATCH --gpus-per-node=1``` because it had a redundancy conflict causing an error with ```#SBATCH --gres=gpu:v100:1```.

**Comment by MarcoKurepa** [comment]
The job is now pending I will update you when it has been completed.

**Comment by hosseinfani** [comment]
@MarcoKurepa 
I think you need to change the email notification to your own. I've been receiving some notification from computecanada :D

**Comment by MarcoKurepa** [comment]
> @MarcoKurepa 
> 
> I think you need to change the email notification to your own. I've been receiving some notification from computecanada :D 

Apologies for that! I'll look into it🙂

**Comment by MarcoKurepa** [comment]
Sorry if you got any more notifs today, I just realized they were coming from the slurm script! You shouldn't receive any more emails now:)

**Comment by hosseinfani** [comment]
@MarcoKurepa no worries. correct, I don't receive them anymore. tnx.

**Comment by rezaBarzgar** [comment]
@MarcoKurepa 
Hey Marco, how are the experiments going on?
Also, Have you started the write-up? 
I just wanted to remind you that these tasks are the priority gently. Please update me regularly even if you have not made any progress or you are busy with some other things.

**Comment by MarcoKurepa** [comment]
> @MarcoKurepa Hey Marco, how are the experiments going on? Also, Have you started the write-up? I just wanted to remind you that these tasks are the priority gently. Please update me regularly even if you have not made any progress or you are busy with some other things.

Hey Reza, sorry for the lack of updates. I have mostly been working on the computecanada cluster, although to be frank I've been having a lot of hiccups with it especially in getting the correct versions of the libraries. At the moment I am still trying to squash errors relating to runnign the script but I have an ```.sh``` ready to run once I've gotten to the point where I can run a sample dataset in the terminal.

I'll make sure to keep you posted moving forward.

**Comment by rezaBarzgar** [comment]
Results' sheets:
https://docs.google.com/spreadsheets/d/1oatYd2uMXOah8RQJsvaT2NIwxhh6CfssaneVLuYczQY/edit#gid=1683104529

**Comment by MarcoKurepa** [comment]
While I was working on the CL Visualization issue, I made an error which resulted in VSCode crashing and I lost the progress made on training CL on the Github dataset. However, in the 48~ hours that it did run ti only finished 2 epoches (still the first fold), so I think it'd be better to run it off Reza's computer regardless. Also I would've needed to restart the run either way to include the expert loss logging script, so no real progress was lost, just a mild annoyance in the end.

@rezaBarzgar

**Comment by hosseinfani** [comment]
#236
